### PR TITLE
Add created_date field to Products

### LINKED ISSUE #5439: PNI - add created_date to Product Updates
We cannot currently filter product updates on age, and we really want to be able to do that in the future.

- [x] add created_date to `Update` class in product_update.py
  - [x] `auto_now=True`
  - [ ] read-only field for admin purposes
  - [x] default datetime of `2020-01-01` as migration safety

### PR DESCRIPTION
**Pull request by KalobTaulien**
Closes #5439 

Simply added a new field. Translations not required on a date field. Field is not required for new products either.

### CODE REVIEW

**Review by Pomax — CHANGES_REQUESTED** [review]
Products already have a review date, the issue was for adding a created_date to Product Updates (found in product_update.py)

**Comment by Pomax** [comment]
I've updated the issue with acceptance criteria.

**Comment by KalobTaulien** [comment]
@Pomax I made those updates. Thanks for the feedback! 

One more quick review from your end?

File: network-api/networkapi/buyersguide/pagemodels/product_update.py [hunk 1]
```diff
@@ -26,5 +30,19 @@ class Update(models.Model):
         blank=True,
     )
 
+    created_date = models.DateField(
+        auto_now=True,
+        null=True,
```
Comment (Pomax) [inline network-api/networkapi/buyersguide/pagemodels/product_update.py:35]: we can actually remove this, and structure the migration as you have it right now, but with a third operation for `AlterField` that removes the `null=True`, as every product update will have a value assigned after `set_default_created_date` runs.

**Review by Pomax — CHANGES_REQUESTED** [review]
Small thing, but not having nulls is always peace of mind.

File: network-api/networkapi/buyersguide/migrations/0007_update_created_date.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='update',
+            name='created_date',
+            field=models.DateField(auto_now=True, null=True, help_text='The date this product was created'),
+        ),
+        migrations.RunPython(set_default_created_date),
+    ]
```
Comment (Pomax) [inline network-api/networkapi/buyersguide/migrations/0007_update_created_date.py:29]:
```suggestion
```suggestion
        migrations.AlterField(
            model_name='update',
            name='created_date',
            field=models.DateField(auto_now=True, help_text='The date this product was created'),
        ),
    ]
```

File: network-api/networkapi/buyersguide/pagemodels/product_update.py (hunk 1 above)
Comment (Pomax) [reply network-api/networkapi/buyersguide/pagemodels/product_update.py:35]:
```suggestion
```

File: network-api/networkapi/buyersguide/migrations/0007_update_created_date.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+    dependencies = [
+        ('buyersguide', '0006_auto_20201015_1607'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='update',
+            name='created_date',
+            field=models.DateField(auto_now=True, null=True, help_text='The date this product was created'),
```
Comment (Pomax) [inline network-api/networkapi/buyersguide/migrations/0007_update_created_date.py:26]:
```suggestion
            field=models.DateField(auto_now_add=True, null=True, help_text='The date this product was created'),
```

File: network-api/networkapi/buyersguide/migrations/0007_update_created_date.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        ),
+        migrations.RunPython(set_default_created_date),
+        migrations.AlterField(
+            model_name='update',
+            name='created_date',
+            field=models.DateField(auto_now=True, help_text='The date this product was created'),
```
Comment (Pomax) [inline network-api/networkapi/buyersguide/migrations/0007_update_created_date.py:32]:
```suggestion
            field=models.DateField(auto_now_add=True, help_text='The date this product was created'),
```

File: network-api/networkapi/buyersguide/pagemodels/product_update.py [hunk 5]
```diff
@@ -26,5 +28,18 @@ class Update(models.Model):
         blank=True,
     )
 
+    created_date = models.DateField(
+        auto_now=True,
```
Comment (Pomax) [inline network-api/networkapi/buyersguide/pagemodels/product_update.py:32]:
```suggestion
        auto_now_add=True,
```

**Review by Pomax — CHANGES_REQUESTED** [review]
https://docs.djangoproject.com/en/2.2/ref/models/fields/#datefield rofflecopters

**Comment by KalobTaulien** [comment]
@Pomax I merged in your suggestions, thank you for those! 

Can I get your approval to merge this in?

**Review by Pomax — APPROVED** [review]
nice!

**Pull request merged**